src/wasm.h

```diff
@@ -1428,6 +1428,7 @@ class Module {
   std::vector<std::unique_ptr<Import>> imports;
   std::vector<std::unique_ptr<Export>> exports;
   std::vector<std::unique_ptr<Function>> functions;
+  std::vector<std::unique_ptr<FunctionType>> externTypes;
```

Comment:
externTypes are the types of functions which are declared but not defined. You're right that it's simpler to keep the types along with the other function types, I fixed that. `externTypesMap` now just maps the declaration name to the type.